dnf-behave-tests/dnf/plugins-core/config-manager-opts.feature

```diff
@@ -0,0 +1,371 @@
+    And file "/etc/dnf/dnf.conf" contents is 
+      """
+      [main]
+      best=True
+      skip_unavailable=1
+      """
+
+
+Scenario: set main option multiple times with the same value is OK
+ Given I create file "/etc/dnf/dnf.conf" with
+      """
+      [main]
+      best=True
+      """
+   When I execute dnf with args "config-manager setopt skip_unavailable=1 best=False skip_unavailable=1"
+   Then the exit code is 0
+    And file "/etc/dnf/dnf.conf" contents is 
+      """
+      [main]
+      best=False
+      skip_unavailable=1
+      """
+
+
+Scenario: set main option more times with different value
+ Given I create file "/etc/dnf/dnf.conf" with
+      """
+      [main]
+      best=True
+      """
+   When I execute dnf with args "config-manager setopt skip_unavailable=1 best=False skip_unavailable=0"
+   Then the exit code is 1
+    And stderr contains "Sets the "skip_unavailable" option again with a different value: "1" != "0""
+
+
+Scenario: setting non-existent main option
+ Given I create file "/etc/dnf/dnf.conf" with
+      """
+      [main]
+      best=True
+      """
+   When I execute dnf with args "config-manager setopt nonexistent=1"
+   Then the exit code is 1
+    And stderr contains "Cannot set option: "nonexistent=1": Option "nonexistent" not found"
+
+
+Scenario: main option - tests for setting an invalid value
+ Given I create file "/etc/dnf/dnf.conf" with
+      """
+      [main]
+      best=True
+      """
+   When I execute dnf with args "config-manager setopt best=XX"
+   Then the exit code is 1
+    And stderr contains "Cannot set option: "best=XX": Invalid boolean value "XX""
+
+
+Scenario: change the value of main option
+ Given I create file "/etc/dnf/dnf.conf" with
+      """
+      [main]
+      best=True
+      skip_unavailable=True
+      """
+   When I execute dnf with args "config-manager setopt best=False"
+   Then the exit code is 0
+    And file "/etc/dnf/dnf.conf" contents is 
+      """
+      [main]
+      best=False
+      skip_unavailable=True
+      """
+
+
+Scenario: unset/remove an main option
+ Given I create file "/etc/dnf/dnf.conf" with
+      """
+      [main]
+      best=True
+      skip_unavailable=True
+      """
+   When I execute dnf with args "config-manager unsetopt best"
+   Then the exit code is 0
+    And file "/etc/dnf/dnf.conf" contents is 
+      """
+      [main]
+      skip_unavailable=True
+      """
+
+
+Scenario: unset/remove an main option, trying to unset an not set option is OK
+ Given I create file "/etc/dnf/dnf.conf" with
+      """
+      [main]
+      best=True
+      skip_unavailable=True
+      """
+   When I execute dnf with args "config-manager unsetopt best installroot"
+   Then the exit code is 0
+    And file "/etc/dnf/dnf.conf" contents is 
+      """
+      [main]
+      skip_unavailable=True
+      """
+
+
+Scenario: repository configuration overrides - new option to "repo1"
+ Given I create directory "/etc/dnf/repos.override.d/"
```

Comment:
I guess that should be additional space.

Reply:
Added